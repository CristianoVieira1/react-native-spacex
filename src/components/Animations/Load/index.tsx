import React from "react";
import { ActivityIndicator } from "react-native";
import Theme from "../../../theme";

function Load() {
  return (
    <ActivityIndicator
      color={Theme.colors.orange}
      style={{
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
      }}
      testID="loading-indicator"
    />
  );
}

export default Load;
