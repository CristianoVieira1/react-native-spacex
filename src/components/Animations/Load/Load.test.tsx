import { render } from "@testing-library/react-native";
import React from "react";
import Load from ".";

describe("Load Component", () => {
  it("renders the loading indicator with the correct color", () => {
    const { getByTestId } = render(<Load />);
    const loadingIndicator = getByTestId("loading-indicator");
    const colorActive = loadingIndicator.props.color;

    expect(colorActive).toBe("#FF8200");
  });

  it("render Load", () => {
    const renderComponent = render(<Load />);
    expect(renderComponent.toJSON()).toMatchSnapshot();
  });
});
