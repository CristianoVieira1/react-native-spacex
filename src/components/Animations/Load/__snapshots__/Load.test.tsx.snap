// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load Component render Load 1`] = `
<ActivityIndicator
  color="#FF8200"
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="loading-indicator"
/>
`;
